### DISCUSSION TITLE
How to upgrade from elasticsearch 7.10.2 helm chart to opensearch operator?

### DISCUSSION THREAD

**Discussion by virasana** [Opster/opensearch-k8s-operator#878] [General]
Hi, 
I am using https://github.com/elastic/elasticsearch  7.10.2.  Is it possible to upgrade to opensearch operator and how?

I am thinking to stand up a new opensearch operator cluster and then shift the elasticsearch data over, followed by pod restarts.
Specifically, the elasticsearch chart has a statefulset for masters, whereas the opensearch operator chart has statefulsets for masters, nodes and co-ordinators.  Should I stand up the elasticsearch masters disks onto the nodes statefulests in the new cluster?

Looking for general advice on how to migrate from elasticsearch 7.10.2 to an opensearch operator cluster, addressing the above concerns.  Are you able to advise?

Many Thanks

**Comment by swoehrl-mw** [comment]
Hi @virasana. An in-place upgrade not possible. The best way would probably be to backup your elasticsearch data, create an opensearch cluster and restore the backup there.

Also see https://opensearch.org/docs/latest/upgrade-to/snapshot-migrate/
